commands2/trapezoidprofilecommand.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+from __future__ import annotations
+
+from typing import TypeVar, Callable, Any, Generic
+
+from wpilib import Timer
+from wpimath.trajectory import TrapezoidProfile, TrapezoidProfileRadians
+
+from .command import Command
+from .subsystem import Subsystem
+
+# Defined two generic types for the Profile and ProfileState variables.
+# This allows an implementation for both dimensionless and Radians
+# instances of TrapezoidProfiles (or any future dimension as well)
+TP = TypeVar("TP")  # Generic[TrapezoidProfile]
```

Comment:
This isn't correct. We want to add a protocol:

```python
class TrapezoidProfile(Protocol, Generic[TS]):
    def calculate(self, t: wpimath.units.seconds, goal: TS, current: TS) -> TS:
        ...
    
    def totalTime(self) -> wpimath.units.seconds:
        ...
```

Then you can do

```python
class TrapezoidProfileCommand(Command, Generic[TS]):
    """
    A command that runs a :class:`.TrapezoidProfile`. Useful for smoothly controlling mechanism motion.

    This class is provided by the NewCommands VendorDep
    """

    def __init__(
        self,
        profile: TrapezoidProfile,
        output: Callable[[TS], Any],
        getGoal: Callable[[], TS],
        getCurrent: Callable[[], TS],
        *requirements: Subsystem,
    )
```

And get rid of those `type: ignore` comments. 

... I feel like the protocol should probably live in wpimath instead of here? @auscompgeek do you have an opinion on that? 

... thinking about it, we probably should have protocols for anything that uses a C++ template, but that sounds obnoxious.

Reply:
I wonder if it would be easier to do this with some kind of covariant/contravariant something or other instead of doing a full-blown protocol.